### PR TITLE
Add command to publish announce message via HTTP

### PR DESCRIPTION
**Pull request by gammazero**
The `announce-http` command manually announces the latest advertisement CID to a specific indexer over HTTP. This works whether the provider publishes advertisements over dtsync or httpsync.

### CODE REVIEW

File: cmd/provider/announce.go [hunk 1]
```diff
@@ -32,3 +43,36 @@ func announceCommand(cctx *cli.Context) error {
...
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return errFromHttpResp(resp)
+	}
+
+	_, err = cctx.App.Writer.Write([]byte("Announced latest advertisement via HTTP\n"))
```
Comment (willscott) [inline cmd/provider/announce.go:76]: maybe include the cid or announcement serialization?

File: engine/host_to_maddr.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	}
+	return ""
+}
+
+func hostToMultiaddr(hostport string) (multiaddr.Multiaddr, error) {
+	host, port, err := net.SplitHostPort(hostport)
+	if err != nil {
+		port = "80"
+		host = hostport
+	}
```
Comment (willscott) [inline engine/host_to_maddr.go:34]: aren't there some forms of hostport that should actually error here?

Comment (gammazero) [reply engine/host_to_maddr.go:34]: Errors should all be caught by `parseAddrType` and `multiaddr.NewMultiaddr`.  I added a unit test that checks failures.

File: cmd/provider/announce.go (hunk 1 above)
Comment (gammazero) [reply cmd/provider/announce.go]: Responds with announced CID.

**Review by willscott — APPROVED** [review]

**Pull request merged**